pilot/pkg/config/clusterregistry/multicluster.go

```diff
@@ -105,8 +105,9 @@ func (m *Multicluster) AddMemberCluster(clientset kubernetes.Interface, clusterI
 
 	m.remoteKubeControllers[clusterID] = &remoteKubeController
 	m.m.Unlock()
-	_ = kubectl.AppendServiceHandler(func(*model.Service, model.Event) { m.XDSUpdater.ConfigUpdate(model.UpdateRequest{Full: true}) })
-	_ = kubectl.AppendInstanceHandler(func(*model.ServiceInstance, model.Event) { m.XDSUpdater.ConfigUpdate(model.UpdateRequest{Full: true}) })
+
+	_ = kubectl.AppendServiceHandler(func(*model.Service, model.Event) { m.updateHandler() })
+	_ = kubectl.AppendInstanceHandler(func(*model.ServiceInstance, model.Event) { m.updateHandler() })
```

Comment:
why does multicluster do a full push when a new instance comes? shouldn't this be a EDS only push

Reply:
This is just a build fix. But as you say, should have a try.  Will do it in a following up.

Reply:
Yeah nothing related to this pr, just something I noticed

Reply:
Actually, it does incremental EDS only, the func passed to it doesnot take effect at all.

See `AppendInstanceHandler` in kube/controller/controller.go